index.js

```diff
@@ -5,14 +5,33 @@ const listData = [
   {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
 ]
 
-listData.forEach(data => {
-  const li = document.createElement('li');
-  const a = document.createElement('a');
-  a.textContent = data.text;
-  a.href = data.to;
-  const img = document.createElement('img');
-  img.src = data.img;
-  img.alt = data.alt;
-  fragment.appendChild(li).appendChild(a).insertBefore(img, a.firstChild);  
-});
-ul.appendChild(fragment);
+const getListData = ()=>{
+  return new Promise(function(resolve, reject){
+    try {
+      resolve(listData);
+    } catch (error) {
+      reject('error') 
+    }
+  })
+}
+
+const createList = ()=> {
+  getListData()
+  .then((data)=>{
+    listData.forEach(data => {
+      const li = document.createElement('li');
+      const a = document.createElement('a');
+      a.textContent = data.text;
+      a.href = data.to;
+      const img = document.createElement('img');
+      img.src = data.img;
+      img.alt = data.alt;
+      fragment.appendChild(li).appendChild(a).insertBefore(img, a.firstChild);  
+    });
+    ul.appendChild(fragment);
+  })
+  .catch(()=>{
+    console.log(reject);
+  })
```

Comment:
課題とそれるのですがここを使うのであれば引数に errorオブジェクトを描いて、
```js
console.log(error);
```
です

```suggestion
  .catch((error)=>{
    console.log(error);
  })
```

Reply:
覚えておいて欲しいのは
thenの第一引数に成功時、第二引数に失敗時の処理をかけます
```js
then(() =>{
 // 成功時の処理
}, () => {
　// 失敗時の処理
})
```
とも書けますよ

Reply:
`  .catch((error)=>{    console.log(error);  })`
確かに、ここのcatchのコールバック関数の引数に何も渡してないと何も表示できないですよね。

あと、catchを使わなくても書けるのですね！覚えておきます！
(そもそもですが、今回の課題ではgetListData().catchの部分の記述は要らなかったかもと先生のコメントで今気づきました..失敗時の処理も必ず書かなきゃいけないと思い込んでいました汗)

教えていただいて気がつくことがたくさんです。。
ありがとうございますm(_ _)m)